app/models/topic.rb

```diff
@@ -621,6 +621,12 @@ def update_status(status, enabled, user, opts = {})
     TopicStatusUpdater.new(self, user).update!(status, enabled, opts)
     DiscourseEvent.trigger(:topic_status_updated, self, status, enabled)
 
+    if status == 'closed'
+      StaffActionLogger.new(user).log_topic_closed(self, closed: enabled)
+    elsif status == 'archived'
+      StaffActionLogger.new(user).log_topic_archived(self, archived: enabled)
```

Comment:
I think we should update the tests for Topic#update_status to ensure that the staff action is logged.

Reply:
Done